### PR TITLE
[stable/coredns] add back the service account for autoscaler

### PR DESCRIPTION
**Pull request by Smirl**
#### Is this a new chart
No

#### What this PR does / why we need it:

- In a previous PR, I remove the `ServiceMonitor` file for the `autoscaler`.
- https://github.com/helm/charts/pull/20430
- however, the file was named incorrectly, and I actually removed the `ServiceAccount` 😿 
- this adds the file back, with the correct name.

#### Which issue this PR fixes
- I haven't raised one

#### Special notes for your reviewer:
- @mrueg do you want to merge your PR before this one?

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/coredns/templates/serviceaccount-autoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.autoscaler.enabled }}
```
Comment (mrueg) [inline stable/coredns/templates/serviceaccount-autoscaler.yaml]: This probably should check for rbac enabled as well

Comment (Smirl) [reply stable/coredns/templates/serviceaccount-autoscaler.yaml]: good spot

**Pull request merged**